tests/unittests/InterpreterTest.cpp

```diff
@@ -898,6 +898,90 @@ TEST(Interpreter, nonLinearClassifier) {
   }
 }
 
+/// Generate images in two classes. A "line" is labeled as 0
+/// and a "cross" is labeled as 1
+static void generateImageData(Tensor &images, Tensor &labels, unsigned numSamples, unsigned isTrain) {
+  auto L = labels.getHandle<size_t>();
+  auto image = images.getHandle<>();
+  images.zero();
+  
+  for (size_t i = 0; i < numSamples; i++) {
+    bool isLine = i % 2 == 0;
+    L.at({i, 0}) = isLine ? 0 : 1;
+    size_t target;
+    // to make sure testing set is not totally included in 
+    // training set
+    if (isTrain)  {
```

Comment:
Why is this necessary? `rand()` will not return the same values across different calls to `generateImageData()`.

Reply:
Since the number of iteration and numSamples is large, all possible lines and crosses are train without this condition. This isTrain just want to make sure that we generate some images only used for testing